system-x/services/aws/s3/src/main/java/software/tnb/aws/s3/service/Minio.java

```diff
@@ -0,0 +1,221 @@
+package software.tnb.aws.s3.service;
+
+import software.tnb.aws.common.account.AWSAccount;
+import software.tnb.aws.common.service.AWSService;
+import software.tnb.aws.s3.validation.S3Validation;
+import software.tnb.common.account.Accounts;
+import software.tnb.common.config.OpenshiftConfiguration;
+import software.tnb.common.deployment.OpenshiftDeployable;
+import software.tnb.common.deployment.WithDockerImage;
+import software.tnb.common.deployment.WithExternalHostname;
+import software.tnb.common.deployment.WithInClusterHostname;
+import software.tnb.common.deployment.WithName;
+import software.tnb.common.openshift.OpenshiftClient;
+import software.tnb.common.utils.IOUtils;
+import software.tnb.common.utils.WaitUtils;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import com.google.auto.service.AutoService;
+
+import java.net.URI;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import cz.xtf.core.openshift.helpers.ResourceFunctions;
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.kubernetes.api.model.ContainerPortBuilder;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.IntOrString;
+import io.fabric8.kubernetes.api.model.PersistentVolumeClaimBuilder;
+import io.fabric8.kubernetes.api.model.Quantity;
+import io.fabric8.kubernetes.api.model.ServiceBuilder;
+import io.fabric8.kubernetes.api.model.ServicePortBuilder;
+import io.fabric8.kubernetes.api.model.ServiceSpecBuilder;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
+import io.fabric8.kubernetes.client.PortForward;
+import io.fabric8.openshift.api.model.RouteBuilder;
+import io.fabric8.openshift.api.model.RoutePortBuilder;
+import io.fabric8.openshift.api.model.RouteTargetReferenceBuilder;
+import software.amazon.awssdk.services.s3.S3Client;
+
+@AutoService(Minio.class)
+public class Minio extends AWSService<AWSAccount, S3Client, S3Validation> implements
+    OpenshiftDeployable, WithInClusterHostname, WithName, WithDockerImage, WithExternalHostname {
```

Comment:
you need to split the deployments into local and openshift variants (for example like in https://github.com/tnb-software/TNB/tree/main/system-x/services/fhir/src/main/java/software/tnb/fhir/resource) (I know the local deployment isn't valid for rhoc, but we should always do both variants if it is possible)